### ISSUE TITLE
Mopidy does not recover from internal IP change.

### ISSUE THREAD

**Issue by VGrol** [mopidy/mopidy-gmusic#201]
Salutations,

Mopidy seems to have trouble with an OpenVPN change.
If the VPN configuration/connection is replaced while Mopidy is running and streaming music, 
it will end up failing on the next 'fetch', be that from automatic song transition or a command.
It should be noted that Openvpn does add internal IP routes and the 'internal IP' does change.

It should be noted that once it falls over, it will still 'function', as in, mopidy itself remains active but any incoming connections are timed out or refused.
I personally use the mopidy-mpris and mpd modules with an ncmpcpp client, upon failure without any verbose flags it just keeps repeating 

```
INFO     New MPD connection from [::1]:47746
INFO     New MPD connection from [::1]:47750
```

Once it runs out of connections,
I'll just default to rejecting.

```
WARNING  Rejected connection from [::1]:47848
WARNING  Rejected connection from [::1]:47850
```

with the port number changing, as they're dynamically allocated of course.
It should be noted that using 'killall mopidy' or killing the console which runs mopidy at this point will get it to try to exit, however it will remain an active process and linger.

```
INFO     GLib mainloop got SIGTERM. Exiting...
INFO     Stopping Mopidy frontends
```

However when running Mopidy with verbose output (just tested -vvv) it spits out a little bit more.
As is visible from the end of the code, it'll do the whole 'new connection thing' as described above almost instantly.
```

DEBUG    2018-08-26 14:43:04,407 [13278:MpdSession-11] mopidy.mpd.session
  Request from [::1]:49226: idle
DEBUG    2018-08-26 14:43:05,247 [13278:Dummy-14] mopidy.audio.gst
  Got about-to-finish event.
DEBUG    2018-08-26 14:43:05,247 [13278:Dummy-14] mopidy.audio.actor
  Running about-to-finish callback.
DEBUG    2018-08-26 14:43:05,250 [13278:GMusicBackend-4] gmusicapi.Mobileclient1
  GetStreamUrl(args=[u'[redacted], '[redacted]', u'hi'], kwargs={})
DEBUG    2018-08-26 14:43:05,408 [13278:MpdSession-11] mopidy.mpd.session
  Request from [::1]:49226: noidle
DEBUG    2018-08-26 14:43:05,408 [13278:MpdSession-11] mopidy.mpd.session
  Response to [::1]:49226: OK
DEBUG    2018-08-26 14:43:05,409 [13278:MpdSession-11] mopidy.mpd.session
  Request from [::1]:49226: status
DEBUG    2018-08-26 14:43:15,419 [13278:MainThread] pykka
  Registered MpdSession (urn:uuid:4e8e581d-fc19-4f05-8ba4-3e5123e6f86a)
DEBUG    2018-08-26 14:43:15,419 [13278:MainThread] pykka
  Starting MpdSession (urn:uuid:4e8e581d-fc19-4f05-8ba4-3e5123e6f86a)
INFO     2018-08-26 14:43:15,420 [13278:MpdSession-15] mopidy.mpd.session
  New MPD connection from [::1]:49412
DEBUG    2018-08-26 14:43:15,420 [13278:MpdSession-15] mopidy.mpd.session
  Request from [::1]:49412: status
DEBUG    2018-08-26 14:43:25,429 [13278:MainThread] pykka
  Registered MpdSession (urn:uuid:ab7b4a53-9661-4efc-ba90-743eeefb22fe)
DEBUG    2018-08-26 14:43:25,429 [13278:MainThread] pykka
  Starting MpdSession (urn:uuid:ab7b4a53-9661-4efc-ba90-743eeefb22fe)
INFO     2018-08-26 14:43:25,430 [13278:MpdSession-16] mopidy.mpd.session
  New MPD connection from [::1]:49426
DEBUG    2018-08-26 14:43:25,430 [13278:MpdSession-16] mopidy.mpd.session
  Request from [::1]:49426: status
```

Basically, It falls flat on its face if I change the OpenVPN profile, I think this might have to do with the internal IP routes which change, but I'm not too certain.
I'd like it to recover somehow, so that I don't have to manually kill the process and restart it.

I run Mopidy locally, as the server added too many variables into the mix.

> Mopidy version 2.1.0
> Mopidy Gmusic version 3.0.0
> Mopidy Mpris version 1.3.1
> Mopidy backends: M3UBackend, GmusicBackend
> Mopidy frontends: Mpdfrontend, MprisFrontend, GmusicScrobblerFrontend
> ncmpcpp version 0.8.2

Running on [localhost], I've also tried my internal network ip, 127.0.0.1, (as local host defaults to ::1), variants of the IP route and variants of 127.0.0.1.

**Comment by adnidor** [comment]
This sounds like an issue with the mopidy core and should be tracked on that repository.

**Comment by VGrol** [comment]
> This sounds like an issue with the mopidy core and should be tracked on that repository.

I wasn't exactly certain if it'd be an MPD/Mpris problem or Gmusic, since It'd probably work with local files. Do you need me to open an instance there or will it just get referenced?

**Comment by jodal** [comment]
Closing because Google Play Music has been shut down, and this project is being discontinued.
